File: backend/src/main/java/com/example/model/Review.java
package com.example.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

/** 商品レビューのドメインクラス. */
@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "reviews")
public class Review {
  /** レビューID. */
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "review_id")
  private Integer id;

  /** コメント. */
  @Column(name = "comment", nullable = false, length = 255)
  private String comment;

  /** 評価（1～5）. */
  @Column(name = "rating", nullable = false)
  @ColumnDefault("1")
  private Integer rating;

  /** 商品カテゴリ（0:PC, 1:Book）. */
  @Column(name = "product_category", nullable = false)
  private Integer productCategory;

  /** 商品ID. */
  @Column(name = "product_id", nullable = false)
  private Integer productId;

  /** ユーザー. */
  @ManyToOne
  @JoinColumn(name = "user_id", nullable = false)
  private User user;
}

File: frontend/src/features/product/types/Review.ts
/** レビューの型定義 */
export type Review = {
  /** レビューID */
  id: number;
  /** コメント */
  comment: string;
  /** 評価（1～5） */
  rating: 1 | 2 | 3 | 4 | 5;
  /** 商品カテゴリ（0:PC, 1:Book） */
  productCategory: number;
  /** 商品ID */
  productId: number;
  /** ユーザーID */
  userId: number;
  /** ユーザー名 */
  userName: string;
};

export type ReviewCounts = Record<Review["rating"], number>;
